### PR TITLE
Update Boost to v1.74.0

### PR DESCRIPTION
**Pull request by Bjoe**
* I've followed [this guide](https://hunter.readthedocs.io/en/latest/creating-new/update.html)
  step by step carefully. **Yes**

### CODE REVIEW

File: cmake/projects/Boost/hunter.cmake [hunk 1]
```diff
@@ -181,6 +181,17 @@ hunter_add_version(
     04f570acbe0beb762e588ad3de292d0328a79c64
 )
 
+hunter_add_version(
+    PACKAGE_NAME
+    Boost
+    VERSION
+    "1.74.0-p0"
+    URL
+    "https://github.com/Bjoe/hunter-boost/archive/hunter-1.74.0.zip"
```
Comment (Bjoe) [inline cmake/projects/Boost/hunter.cmake:190]: If CI builds are all fine, then I will create a PR in Boost...

**Comment by Bjoe** [comment]
@rbsheth I fixed some issue to build boost but somehow the github action didn't trigger the ci builds again ... I assume that the ci builds again when I finishing this PR .... not sure

**Comment by Bjoe** [comment]
I''m struggling with the CI builds ... how can I restarts the Ci builds? At least the failing jobs? Also on my fork it doesn't work. Github actions are "on". Or has this something to do that have more then  256 jobs per workflow? I added for the other Boost examples a job....

**Comment by rbsheth** [comment]
> I''m struggling with the CI builds ... how can I restarts the Ci builds? At least the failing jobs? Also on my fork it doesn't work. Github actions are "on". Or has this something to do that have more then 256 jobs per workflow? I added for the other Boost examples a job....

Probably - try just the basic builds first. We can split project matrices to different folders if needed.

**Comment by Bjoe** [comment]
@rbsheth Ok I reduce the toolchain matrix, now it builds again :-). But let try to do some last fixes. I will give you a hint when I'm ready

**Comment by rbsheth** [comment]
Sounds good!

File: cmake/projects/Boost/hunter.cmake [hunk 2]
```diff
@@ -442,7 +442,7 @@ if(MSVC)
   )
 endif()
 
-if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.3)
+if(NOT HUNTER_Boost_VERSION VERSION_LESS 1.72.0 AND CMAKE_VERSION VERSION_GREATER_EQUAL 3.3)
```
Comment (NeroBurner) [inline cmake/projects/Boost/hunter.cmake:446]: could you add some brackets, so I don't have to look up the boolean rules if the `NOT` is evaluated before or after the `AND`

```cmake
if( (NOT HUNTER_Boost_VERSION VERSION_LESS 1.72.0) AND (CMAKE_VERSION VERSION_GREATER_EQUAL 3.3) )
```

Comment (Bjoe) [reply cmake/projects/Boost/hunter.cmake]: Fixed.

Comment (NeroBurner) [reply cmake/projects/Boost/hunter.cmake]: thanks!

**Comment by rbsheth** [comment]
@Bjoe Happy new year! How is this going?

**Comment by Bjoe** [comment]
> @Bjoe Happy new year! How is this going?

Happy new year @rbsheth .... sorry, I try to use the "Boost-Config" from boost project as default like FindBoost in cmake did and to resolve this issue that the "FindBoost" in hunter is outdated. (See discussion #141) Unfortunately the boost project is using `IN_LIST` operator and it needs cmake version 3.3 at least. I try to force or to verify via hunter to set this policy or disable to use boost config. It works but unfortunately, there is a depended project `BoostProcess` and it set as cmake minimum version 3.0 and didn't yet any solution how I can verify in cmake which "minimum version" are set :-(.

So I revert commit eb2aff8 to have still the "old" solution with the option to set explicitly `USE_CONFIG_FROM_BOOST` ...

Also the windows vs-15 builds are broken because of a "space" in a path! I try to solve but it was yet impossible for me. I need a windows machine for this. Maybe I found time to setup one Window machine in a virtualbox ... but this is also a bigger task ....

So let me revert a little bit, cleanup and then merge ....

**Comment by Bjoe** [comment]
Ok when the build is done, this can be merged! Please squash and merge my commits!

File: cmake/projects/Boost/hunter.cmake (hunk 1 above)
Comment (Bjoe) [reply cmake/projects/Boost/hunter.cmake]: Done

**Review by rbsheth — APPROVED** [review]
Thanks!!

**Comment by rbsheth** [comment]
Released: https://github.com/cpp-pm/hunter/releases/tag/v0.23.290

**Comment by olliwang** [comment]
Any idea why this PR breaks iOS build? v0.23.289 works just fine but not since v0.23.290 anymore.

**Pull request merged**